### DISCUSSION TITLE
Question: How to create a Rectangle Chart instead of Bubbles?

### DISCUSSION THREAD

**Discussion by brainbytes42** [GSI-CS-CO/chart-fx#545] [General]
Hi,
I'm constantly running into flaws of the original JavaFX Charts, so I was really happy to find this library; it seems really promising to me, but I'm new to it and try to explore it's features. :-)
But until now, I didn't find any hint for my problem, so maybe someone is kind enough to help? Thanks in advance!

I need a chart somehow similar to a bubble-chart, but not with circular bubbles, but with rectangles as markers. So for each (x,y)-Coordinate, I have an individual length- and width-value (instead of a single radius-value). 

What is the best way to approach this? I don't want to reinvent the wheel, if maybe there is a simple solution but I don't find it... ;-)

**Comment by wirew0rm** [comment]
Hey, thanks for your interests and your kind words :)

At the moment it seems that the bubble shape is hardcoded to be a circle/oval. I would suggest creating a custom renderer by overriding `drawBubbles` in https://github.com/fair-acc/chart-fx/blob/ef5ee2e60a04f1f304826046f3cf3963ba9870e2/chartfx-chart/src/main/java/io/fair_acc/chartfx/renderer/spi/ErrorDataSetRenderer.java#L334. Just copy the method and replace `fillOval` by `fillRect`.

Potentially we could also add a `BubbleStyle` corresponding to `MarkerStyle` and define different predefined shapes/allow to provide a custom method. If you want to provide a Pull request that implements that, feel free to do so and we would happily review and merge that. (Still I would suggest to first implement the override solution to get fast results and a feeling for the functionality).

As for the separate x and y values, that sounds more like a style for (symmetrical) error bars to me. That would mean overriding https://github.com/fair-acc/chart-fx/blob/ef5ee2e60a04f1f304826046f3cf3963ba9870e2/chartfx-chart/src/main/java/io/fair_acc/chartfx/renderer/spi/ErrorDataSetRenderer.java#L411 to draw a box instead of two (capped) lines. That would allow you to use the datasets.

**Comment by brainbytes42** [comment]
Thanks for the quick reply, I'll have a look into that class.
